core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
@guozhangwang Latest commit has added a new `shutdown()` method in `BaseConsumer` that is consistent with the old consumer shutdown. It is idempotent with both old and new consumer, so it can be invoked multiple times. So `shutdown()` is invoked safely from the finally block as well as the shutdown hook without any error logging or exceptions.